src/app/services/tasksService/tasks.service.ts

```diff
@@ -7,85 +7,132 @@ export interface ActionItem {
   type?: string;
   completed?: string;
   dueDate?: Date;
+  id: string;
+}
+
+export interface AddActionItem {
+  title: string;
+  projectName: string;
+  type?: string;
+  completed?: string;
+  dueDate?: Date;
 }
 
 @Injectable({
   providedIn: 'root'
 })
 export class TasksService {
+  justAddedItem: ActionItem;
   private readonly dataTable: ActionItem[] = [
     {
       title: 'Android - UI Automation Test',
       projectName: 'CASD Wilson & Lamberton Middle Schools',
       type: 'General',
       completed: '80',
-      dueDate: new Date('2019/11/17')
+      dueDate: new Date('2019/11/17'),
+      id: '1'
     },
     {
       title: 'The Flash Tutorial',
       projectName: 'CASD Wilson & Lamberton Middle Schools',
       type: 'General',
       completed: '70',
-      dueDate: new Date('2019/12/29')
+      dueDate: new Date('2019/12/29'),
+      id: '2'
     },
     {
       title: 'Cleaning and Organising Your Computer',
       projectName: 'CASD Wilson & Lamberton Middle Schools',
       type: 'Clash',
       completed: '0',
-      dueDate: new Date('2019/11/15')
+      dueDate: new Date('2019/11/15'),
+      id: '3'
     },
     {
       title: 'Android - UI Automation Test',
       projectName: 'CASD Wilson & Lamberton Middle Schools',
       type: 'General',
       completed: '80',
-      dueDate: new Date('2019/11/17')
+      dueDate: new Date('2019/11/17'),
+      id: '4'
     },
     {
       title: 'The Flash Tutorial',
       projectName: 'CASD Wilson & Lamberton Middle Schools',
       type: 'General',
       completed: '70',
-      dueDate: new Date('2019/11/16')
+      dueDate: new Date('2019/11/16'),
+      id: '5'
     },
     {
       title: 'Android - UI Automation Test',
       projectName: 'CASD Wilson & Lamberton Middle Schools',
       type: 'General',
       completed: '80',
-      dueDate: new Date('2019/11/17')
+      dueDate: new Date('2019/11/17'),
+      id: '6'
     },
     {
       title: 'The Flash Tutorial',
       projectName: 'CASD Wilson & Lamberton Middle Schools',
       type: 'General',
       completed: '70',
-      dueDate: new Date('2019/11/16')
+      dueDate: new Date('2019/11/16'),
+      id: '7'
     },
     {
       title: 'Cleaning and Organising Your Computer',
       projectName: 'CASD Wilson & Lamberton Middle Schools',
       type: 'Clash',
       completed: '0',
-      dueDate: new Date('2020/01/05')
+      dueDate: new Date('2020/01/05'),
+      id: '8'
     }
   ];
-
-  add(item: ActionItem): Observable<ActionItem> {
+  add(item: AddActionItem): Observable<ActionItem> {
     return new Observable(observer => {
       setTimeout(() => {
-        this.dataTable.push(item);
-        observer.next(item);
+        const newActionItem = {
+          ...item,
+          id: this.getLastItemId()
+        };
+        this.dataTable.push(newActionItem);
+        observer.next(newActionItem);
+        this.justAddedItem = this.getLastItem();
```

Comment:
**CONCERN:** Why do you need this assignment if you're returning this item as a result?